### PR TITLE
Prompt Crafter json loads fix for numbers

### PR DESCRIPTION
**Pull request by SamGos93**
Fix the failure when output pattern has a number and the json loads fails. 
Removing the try/except logic for json loads as for chat models, we would always need the role to be added.

### CODE REVIEW

**Review by alpatots — APPROVED** [review]

File: assets/aml-benchmark/components/src/prompt_crafter/package/prompt_factory.py [hunk 1]
```diff
@@ -330,6 +330,11 @@ def _parse_affix(self, affix: str, role="user") -> OpenAICreateChatPrompt:
             # NOTE: user role is currently prefered over system role
             # though this may change in the future as models are updated
             messages = [{"role": role, "content": affix}]
+
+        # Handling the case where Json.loads load a string with only number as intger.
+        if isinstance(messages, int):
```
Comment (alpatots) [inline assets/aml-benchmark/components/src/prompt_crafter/package/prompt_factory.py]: Do we need a unit test?

**Review by alpatots — APPROVED** [review]

**Comment by mercerchen** [comment]
@SamGos93 Can you point me to the checksum test pipeline? Tried to find it in the checks but couldn't find it. I am assuming this would be breaking for this test.
image

**Review by mercerchen — CHANGES_REQUESTED** [review]
prompt_factory line 327 to 332 would be breaking for the screenshoted case I put in the issue

File: assets/aml-benchmark/components/src/prompt_crafter/package/prompt_factory.py (hunk 1 above)
Comment (SamGos93) [reply assets/aml-benchmark/components/src/prompt_crafter/package/prompt_factory.py:335]: Added test cases for this.

**Comment by SamGos93** [comment]
> @SamGos93 Can you point me to the checksum test pipeline? Tried to find it in the checks but couldn't find it. I am assuming this would be breaking for this test. image

@mercerchen : I have updated the logic. Can you please check?

**Comment by mercerchen** [comment]
@SamGos93 Can you tell me thne name of the test pipeline? the code looks good just want to see the tests passing

**Comment by SamGos93** [comment]
> @SamGos93 Can you tell me thne name of the test pipeline? the code looks good just want to see the tests passing

It is probably this https://github.com/Azure/azureml-assets/actions/runs/6592757956/job/17914044989?pr=1527. We would know once it completes/ fails.

**Review by alpatots — APPROVED** [review]

**Review by iamrk04 — APPROVED** [review]

**Pull request merged**